### ISSUE TITLE
summary

### ISSUE THREAD

**Issue by cjessett** [shiina-ringo/jammming_feature#1]
Overall, nice work.

You chose a feature that would improve the user experience and you  clearly communicated your solution. However, I would suggest a simpler solution and a pattern that scales well when React apps become very large and you want to filter multiple parts of the state.

In general, when filtering parts of state, you want to do it at the highest level, i.e. the component that is holding the state. In this project that would be the `App` component. This way, instead of creating new props and making sure to pass everything down through all of the components involved, we just have to filter the part of state we want in `App`. 

The part of state we want to filter is the `searchResults`. So, just as you filtered the `tracks` prop that held the `searchResults` in the `Tracklist` component we want to filter the `searchResults` before they are passed to the `SearchResults` component. Then, the only changes we would need to make is the filter function you created, which now goes in the `App` component. And then we want to create a new variable that holds the filtered results. So, we use that filter function to filter `this.state.searchResults` in the `render` function of `App` before the `return`. Finally, we pass the filtered search results, let's call it `filteredSearchResults` to the `SearchResults` component via the `searchResults` prop. 

To sum it up, when filtering parts of state, do the filtering at the highest level, that is, in the component that holds the state. This way, child components only get the minimum they need to function. This keeps the rest of our components simpler, dumber, and cleaner.
